app/assets/javascripts/live_search.js

```diff
@@ -115,6 +115,14 @@
       this.$suggestionsBlock.setAttribute('data-search-query', this.currentKeywords())
     }
     if (GOVUK.Ecommerce) { GOVUK.Ecommerce.start() }
+
+    if (GOVUK.Modules) {
+      var searchResults = document.querySelectorAll("[data-module='gem-track-click ga4-enhanced-ecommerce-tracker']")
+
+      for (var i = 0; i < searchResults.length; i++) {
+        new GOVUK.Modules.Ga4EnhancedEcommerceTracker(searchResults[i]).init(false)
```

Comment:
I haven't looked at the accompanying PR for this yet but I notice that this has been written as a GOVUK Module, which might not be the right approach. Modules are automatically initialised on a page wherever the Modules start script finds a `data-module` attribute, so in this situation it would have already been initialised by the time this code is run.

However I think we might need to re-initialise this ecommerce code after search updates, which is why I suspect this line is here? In which case it might make more sense to not have the new ecommerce code as a Module, but structure it differently somehow. Open to having a conversation about this.

Reply:
Just to document/reflect the discussion we had yesterday - the ecommerce tracking code will no longer be structured as a module so we won't be using this approach any more.